sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/encoder/jsonencoder/_utils.py

```diff
@@ -0,0 +1,268 @@
+        schema_str = schema
+    except TypeError:
+        # callable
+        schema = cast(Callable, schema)
+        try:
+            schema_dict = schema(content)
+        except Exception as exc:
+            raise InvalidContentError(
+                f"Cannot generate schema with callable given the following content: {content}"
+            ) from exc
+        schema_str = json.dumps(schema_dict)
+
+    # get schema name for get_schema_properties operation
+    try:
+        schema_fullname = schema_dict['title']
+    except KeyError:
+        raise ValueError("Schema must have 'title' property.") from None
+
+    return schema_fullname, schema_str, schema_dict
+
+@overload
+def create_message_content(
+    content: Mapping[str, Any],
+    schema: Mapping[str, Any],
+    schema_id: str,
+    validate: "Validator",
+    message_type: Type[MessageType],
+    **kwargs: Any,
+) -> MessageType:
+    ...
+
+@overload
+def create_message_content(
+    content: Mapping[str, Any],
+    schema: Mapping[str, Any],
+    schema_id: str,
+    validate: "Validator",
+    message_type: None = None,
+) -> MessageContent:
+    ...
+
+def create_message_content(
+    content: Mapping[str, Any],
+    schema: Mapping[str, Any],
+    schema_id: str,
+    validate: Union["Validator", "SchemaContentValidate"],
+    message_type: Optional[Type[MessageType]] = None,
+    **kwargs: Any,
+) -> Union[MessageType, MessageContent]:
+    content_type = f"{JSON_MIME_TYPE}+{schema_id}"
+    try:
+        # validate content
+        validate(schema=schema, content=content)
+    except Exception as exc:  # pylint:disable=broad-except
+        raise InvalidContentError(
+            f"Invalid content value '{content}' for the following schema with schema ID {schema_id}:"
+            f"{json.dumps(schema)}",
+            details={"schema_id": f"{schema_id}"},
+    ) from exc
+
+    try:
+        content_bytes = json.dumps(content, separators=(',', ':')).encode()
+    except Exception as exc:
+        raise InvalidContentError(
+            f"Cannot encode value '{content}' for the following schema with schema ID {schema_id}:"
+            f"{json.dumps(schema)}",
+            details={"schema_id": f"{schema_id}"},
+        ) from exc
+
+    if message_type:
+        try:
+            return cast(
+                MessageType,
+                message_type.from_message_content(content_bytes, content_type, **kwargs)
+            )
+        except AttributeError as exc:
+            raise TypeError(
+                f"""Cannot set content and content type on model object. The content model
+                    {str(message_type)} must be a subtype of the MessageType protocol.
+                    If using an Azure SDK model class, please check the README.md for the full list
+                    of supported Azure SDK models and their corresponding versions.""",
+                {"content": content_bytes, "content_type": content_type},
+            ) from exc
```

Comment:
will the AttributeError have info that is useful or should it be None ?

Reply:
I think it's useful. It'll say that `message_type` class has no attribute `from_message_content`, which clarifies which message_type they pass in and the method that it's missing to actually be a "MessageType" subtype.